stingray/simulator/simulator.py

```diff
@@ -134,13 +132,13 @@ def simulate(self, *args):
             lightCurve: `LightCurve` object
         """
 
-        if type(args[0]) == int:
+        if type(args[0]) == int and len(args) == 1: 
             return  self._simulate_power_law(args[0])
 
         elif len(args) == 1:
             return self._simulate_power_spectrum(args[0])
 
-        elif type(args[0]) == str:
+        elif type(args[0]) == str and len(args) == 2:
             return self._simulate_model(args[0], args[1])
```

Comment:
Same here. One way might be using six:
```
import six
if isinstance(args[0], six.string_types):
    (....)
```

But six internally does just the following:
```
import sys
PY3 = sys.version_info[0] == 3

if PY3:
    string_types = str,
else:
    string_types = basestring,
```
so we might just want to copy this and avoid importing six.
Taken [from here](http://stackoverflow.com/questions/1303243/how-to-find-out-if-a-python-object-is-a-string)